### DISCUSSION TITLE
How can I use java to connect to the coordinator on another server

### DISCUSSION THREAD

**Discussion by LinYuHeng-Con** [trinodb/trino#13387] [General]
How can I use java to connect to the coordinator on another server

**Comment by tangjiangling** [comment]
Depending on what you want to do, you can use trino-cli or trino-jdbc if you want to execute a query, or you can use the rest interface if you want to get some status information about the query.

**Comment by LinYuHeng-Con** [comment]
I want to get some information about the query, such as query progress, execution status and other operations. Personally, I want to use the rest API, but there will be problems when connecting to the coordinator

![5B0D8F05-E820-497f-BFFD-D47BE4A79AA2](https://user-images.githubusercontent.com/48077349/180139169-e5834bcf-d513-4647-b98d-049f6b3b74f1.png)

![FC28D70C-3BD2-4231-A58E-72B7E815CDCE](https://user-images.githubusercontent.com/48077349/180139277-79d5d5d6-dcf5-4f39-8b58-ff45ea0919b9.png)

**Comment by hashhar** [comment]
Discovery URI pointing to other coordinator simply registers multiple coordinators in the cluster and will cause issues unless you know what you are doing.

Also TestingTrinoServer is not built for what you are trying to do so you'll run into lots of issues even if you fix this one. The JDBC driver supports what you are asking for.

For just getting stats about query execution you can use the JDBC driver itself - which exposes the query stats (search for `QueryStats` under `client/trino-jdbc`).

**Reply by LinYuHeng-Con** [reply]
The only thing I want to do is to use java to connect the coordinators of another server, and at the same time carry out query execution plans, execute queries and other operations. I wrote my own demo based on the case on the official website

![A15222E2-4862-456f-836C-B7B4B80D9163](https://user-images.githubusercontent.com/48077349/181435815-a5ce8f63-c8a6-4d4b-9716-ba5d86abea30.png)

![068B1B8B-961A-4863-B39F-F4FEE2AE7840](https://user-images.githubusercontent.com/48077349/181435816-a60ddf15-51c4-4fea-8762-556c5f499add.png)

![D8F7210D-C784-4e79-A24D-AB6004E0B53B](https://user-images.githubusercontent.com/48077349/181435826-f70aab8e-383f-42c9-bc0a-b0345f15a245.png)

![75EA7014-C334-4f43-AD38-FA5381FE7670](https://user-images.githubusercontent.com/48077349/181435836-9033b768-c106-4c6a-b627-dc55896d9bf0.png)
